packages/payment/src/loaders/defaults.ts

```diff
@@ -1,10 +1,32 @@
-import { IPaymentModuleService, LoaderOptions } from "@medusajs/types"
 import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import {
+  CreatePaymentProviderDTO,
+  IPaymentModuleService,
+  LoaderOptions,
+} from "@medusajs/types"
 
 export default async ({ container }: LoaderOptions): Promise<void> => {
   const paymentModuleService: IPaymentModuleService = container.resolve(
     ModuleRegistrationName.PAYMENT
   )
 
-  await paymentModuleService.createProvidersOnLoad()
+  const providersToLoad = container.resolve("payment_providers")
+  const paymentProviderService = container.resolve("paymentProviderService")
+
+  const providers = await paymentModuleService.listPaymentProviders({
```

Comment:
nit: we can list these with `paymentProviderService` so we don't need to resolve `paymentModuleService` in the loader